### PR TITLE
[MIG][16.0] stock_picking_group_by_partner_by_carrier

### PR DESCRIPTION
**Pull request by rousseldenis**
Depends on :

- https://github.com/odoo/odoo/pull/108452

### CODE REVIEW

File: stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        <xpath
+            expr='//table[@name="stock_move_table"]/tbody/t[@t-set="lines"]'
+            position="attributes"
+        >
+            <attribute name="t-value">report_lines</attribute>
+        </xpath>
+
+        <xpath
+            expr='//table[@name="stock_move_table"]/tbody/tr//span[@t-field="move.product_id"]/..'
+            position="replace"
```
Comment (rousseldenis) [inline stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml]: @TDU @jbaudoux I don't like replaces but I haven't found a better method at this time

File: stock_picking_group_by_partner_by_carrier/models/stock_move.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            return domain
+
+        # remove group
+        domain = [x for x in domain if x[0] != "group_id"]
+
+        grouping_domain = self._assign_picking_group_domain()
+
+        return domain + grouping_domain
+
+    # TODO: this part and everything related to generic grouping
```
Comment (rousseldenis) [inline stock_picking_group_by_partner_by_carrier/models/stock_move.py:74]: @simahawk @jbaudoux What do you think of this ?

How choosing the good domain as I suppose several group_by_... modules can be installed together

File: stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml (hunk 1 above)
Comment (simahawk) [reply stock_picking_group_by_partner_by_carrier/report/report_delivery_slip.xml]: hide it setting attribute `t-if=False` then add your later

File: stock_picking_group_by_partner_by_carrier/models/stock_move.py (hunk 2 above)
Comment (simahawk) [reply stock_picking_group_by_partner_by_carrier/models/stock_move.py:74]: Yeah, you can have more than one. So far, with the ones available today we didn't have problems.
If you need to take full control, you should probably override this in your custom module.
The idea here is to have a base module for defining an hook and some sort of api for this feat.

**Comment by lmignon** [comment]
@rousseldenis 
Also depends on
 * [x] #1158
 * [x] #1187
> Depends on :
> 
>     * [[FIX] delivery: Really use propagate_carrier option on stock rules odoo/odoo#108452](https://github.com/odoo/odoo/pull/108452)

**Review by lmignon — APPROVED** [review]
Thank you @rousseldenis LGTM (Code review + functional tests)

**Review by FrancoMaxime — APPROVED** [review]
code review

**Comment by sebalix** [comment]
@rousseldenis the dependency has been merged, you can drop your `[DONT MERGE]` commit

**Review by jbaudoux — APPROVED** [review]
@rousseldenis Can you squash the "DO NOT MERGE" commits?

**Comment by jbaudoux** [comment]
@rousseldenis @glitchov Can you drop the "DO NOT MERGE" commit and its revert ? Then we can merge this PR

**Comment by hparfr** [comment]
@rousseldenis is it ready for  merge ?

**Comment by rousseldenis** [comment]
> @rousseldenis is it ready for merge ?

Of course

**Comment by rousseldenis** [comment]
@simahawk Any more comment ?

**Review by cyrilmanuel — APPROVED** [review]

File: stock_quant_package_dimension/README.rst [hunk 3]
```diff
@@ -7,9 +7,9 @@ Stock Quant Package Dimension
    !! changes will be overwritten.                   !!
```
Comment (simahawk) [inline stock_quant_package_dimension/README.rst]: The change in [stock_quant_package_dimension](https://github.com/OCA/stock-logistics-workflow/pull/1165/files#diff-5cc932375429134f87e8cf556c589feeb173fa168716649e0c63036f08305fe4) seems not relevant.

**Review by simahawk — APPROVED** [review]
LG

There's a typo in the commit msg "Use attirbutes instead..." s/attirbutes/attributes.
Also the dev state has been changed in that commit... was it by decision?

**Pull request merged**